### PR TITLE
Feature/task 2 webpack

### PR DESCRIPTION
**Pull request by vposidailo**

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -5,20 +5,34 @@
...
+    ]
   },
   "author": "Vitalii Posidailo",
   "license": "ISC",
+  "babel": {
+    "presets": [
+      "@babel/preset-env",
+      "@babel/preset-react"
+    ]
+  },
```
Comment (EvgeniyTT) [inline frontend/package.json]: You probably don't need to specify babel both in `package.json` and `.babelrc` files. One place is enough.

File: frontend/package.json [hunk 2]
```diff
@@ -5,20 +5,34 @@
...
   "dependencies": {
     "babel-cli": "^6.26.0",
-    "babel-core": "^6.26.3",
-    "babel-loader": "^8.0.5",
     "babel-preset-env": "^1.7.0",
-    "babel-preset-es2015": "^6.24.1",
-    "babel-preset-react": "^6.24.1",
+    "express": "^4.16.4",
```
Comment (EvgeniyTT) [inline frontend/package.json]: Regression :)

File: frontend/package.json [hunk 3]
```diff
@@ -5,20 +5,34 @@
   "main": "index.js",
   "scripts": {
     "start": "react-scripts start",
```
Comment (EvgeniyTT) [inline frontend/package.json]: Let's now start with webpack-dev-server?

File: frontend/package.json [hunk 4]
```diff
@@ -5,20 +5,34 @@
...
-    "babel-loader": "^8.0.5",
     "babel-preset-env": "^1.7.0",
-    "babel-preset-es2015": "^6.24.1",
-    "babel-preset-react": "^6.24.1",
+    "express": "^4.16.4",
     "react": "^16.8.4",
     "react-dom": "^16.8.4",
-    "react-scripts": "^2.1.8"
+    "react-redux": "^6.0.1",
+    "react-router-redux": "^4.0.8",
```
Comment (EvgeniyTT) [inline frontend/package.json]: We probably need `react-router`, not `react-router-redux`
https://www.npmjs.com/package/react-router

**Review by EvgeniyTT — CHANGES_REQUESTED** [review]
I don't see these changes from the task description: "Set DEV and PROD build configuration. Use env variables, dev server, optimizations for PROD build"

File: frontend/package.json (hunk 3 above)
Comment (vposidailo) [reply frontend/package.json]: Done.

File: frontend/package.json (hunk 4 above)
Comment (vposidailo) [reply frontend/package.json]: I am not sure, I will check but I think it was one of task requirements.

File: frontend/package.json (hunk 2 above)
Comment (vposidailo) [reply frontend/package.json]: Done. Removed it. Incorrect merging.

File: frontend/package.json (hunk 1 above)
Comment (vposidailo) [reply frontend/package.json]: Done. Removed from package.json

Comment (EvgeniyTT) [reply frontend/package.json]: Looks like you haven't pushed this change.

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1,3 @@
 frontend/node_modules/
 backend/node_modules/
+frontend/dist/
```
Comment (EvgeniyTT) [inline .gitignore:3]: in real life it makes sense to have two separate `.gitignore` files for frontend and backend. It makes it easier to manage them. And this global/root `.gitignore ` manages global ignores (envs/infrastructure/db/etc.)

File: frontend/__tests__/state-functions.test.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+describe('test', () => {
```
Comment (EvgeniyTT) [inline frontend/__tests__/state-functions.test.js]: small things, just JS-world styles :)
1. test folder is just `test`, no need in underscores. What the meaning they bring?
2. `.test.` usually is called `.spec.` 
3. generally tests are placed in the same directory where the component is, so you have a folder `SomeComponent` and there

```
SomeComponent.js
SomeComponent.css(.scss/.less)
SomeComponent.spec.js
```

Comment (vposidailo) [reply frontend/__tests__/state-functions.test.js]: Sure. It was in some article and I just followed it. Will be fixed in next push.

File: .gitignore (hunk 5 above)
Comment (vposidailo) [reply .gitignore:3]: Could we keep as it is?

File: frontend/package.json (hunk 1 above)
Comment (vposidailo) [reply frontend/package.json]: Pushed all latest changes into the branch.

File: .gitignore (hunk 5 above)
Comment (EvgeniyTT) [reply .gitignore:3]: As you wish.

**Review by EvgeniyTT — APPROVED** [review]
LGTM

**Pull request merged**